packages/components/checkbox/src/checkbox-group.vue

```diff
@@ -1,85 +1,70 @@
-<script lang="ts">
-import {
-  computed,
-  defineComponent,
-  h,
-  nextTick,
-  provide,
-  renderSlot,
-  toRefs,
-  watch,
-} from 'vue'
+<template>
+  <component
+    :is="tag"
+    :id="groupId"
+    :class="ns.b('group')"
+    role="group"
+    :aria-label="!isLabeledByFormItem ? label || 'checkbox-group' : undefined"
+    :aria-labelledby="isLabeledByFormItem ? elFormItem.labelId : undefined"
+  >
+    <slot />
+  </component>
+</template>
+
+<script lang="ts" setup>
+import { computed, nextTick, provide, toRefs, watch } from 'vue'
 import { UPDATE_MODEL_EVENT } from '@element-plus/constants'
 import { debugWarn } from '@element-plus/utils'
 import { useNamespace, useSize } from '@element-plus/hooks'
 import {
+  checkboxGroupEmits,
+  checkboxGroupProps,
   useCheckboxGroup,
   useCheckboxGroupId,
-  useCheckboxGroupProps,
-} from './useCheckbox'
+} from './checkbox-group'
 
-export default defineComponent({
+defineOptions({
   name: 'ElCheckboxGroup',
+})
 
-  props: useCheckboxGroupProps,
-
-  emits: [UPDATE_MODEL_EVENT, 'change'],
-
-  setup(props, { emit, slots }) {
-    const { elFormItem } = useCheckboxGroup()
-    const { groupId, isLabeledByFormItem } = useCheckboxGroupId(props, {
-      elFormItem,
-    })
-    const checkboxGroupSize = useSize()
-    const ns = useNamespace('checkbox')
-
-    const changeEvent = (value) => {
-      emit(UPDATE_MODEL_EVENT, value)
-      nextTick(() => {
-        emit('change', value)
-      })
-    }
+const props = defineProps(checkboxGroupProps)
+const emit = defineEmits(checkboxGroupEmits)
 
-    const modelValue = computed({
-      get() {
-        return props.modelValue
-      },
-      set(val) {
-        changeEvent(val)
-      },
-    })
+const { elFormItem } = useCheckboxGroup()
+const { groupId, isLabeledByFormItem } = useCheckboxGroupId(props, {
+  elFormItem,
+})
+const checkboxGroupSize = useSize()
+const ns = useNamespace('checkbox')
 
-    provide('CheckboxGroup', {
-      name: 'ElCheckboxGroup',
-      modelValue,
-      ...toRefs(props),
-      checkboxGroupSize,
-      changeEvent,
-    })
+const changeEvent = (value) => {
+  emit(UPDATE_MODEL_EVENT, value)
+  nextTick(() => {
+    emit('change', value)
+  })
+}
 
-    watch(
-      () => props.modelValue,
-      () => {
-        elFormItem.validate?.('change').catch((err) => debugWarn(err))
-      }
-    )
-    return () => {
-      return h(
-        props.tag,
-        {
-          id: groupId.value,
-          class: ns.b('group'),
-          role: 'group',
-          'aria-label': !isLabeledByFormItem.value
-            ? props.label || 'checkbox-group'
-            : undefined,
-          'aria-labelledby': isLabeledByFormItem.value
-            ? elFormItem.labelId
-            : undefined,
-        },
-        [renderSlot(slots, 'default')]
-      )
-    }
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+const modelValue = computed({
```

Comment:
@emojiiii Bro, please help to confirm, what is the usage scenario of this `modelValue`? I have thought for a long time that only external modification of this `modelValue` will trigger the `set`, but this property is not actively `exposed`, and it seems that it cannot be modified.